packages/std/src/math/decimal.rs

```diff
@@ -118,16 +118,30 @@ impl Decimal {
 
     /// Returns the ratio (numerator / denominator) as a Decimal
     pub fn from_ratio(numerator: impl Into<Uint128>, denominator: impl Into<Uint128>) -> Self {
+        match Decimal::checked_from_ratio(numerator, denominator) {
+            Ok(value) => value,
+            Err(CheckedMultiplyRatioError::DivideByZero) => {
+                panic!("Denominator must not be zero")
+            }
+            Err(CheckedMultiplyRatioError::Overflow) => panic!("Multiplication overflow"),
+        }
+    }
+
+    /// Returns the ratio (numerator / denominator) as a Decimal
+    pub fn checked_from_ratio(
+        numerator: impl Into<Uint128>,
+        denominator: impl Into<Uint128>,
+    ) -> Result<Self, CheckedMultiplyRatioError> {
```

Comment:
Let's create a dedicated error type here `CheckedFromRatioError` with the same cases

Reply:
Sure. I was thinking about it, but in the end all issues are actually coming down to `checked_multiply_ratio` implementation anyway.  So I didn't want to double the structure just for the sake.

Reply:
Yeah, I see. But then again, this is implementation detail that can change later.